internal/backend/semtechudp/backend.go

```diff
@@ -511,6 +511,7 @@ func (b *Backend) handleStats(gatewayID lorawan.EUI64, stats gw.GatewayStats) {
 		stats.RxPacketsPerModulation = s.RxPacketsPerModulation
 		stats.TxPacketsPerModulation = s.TxPacketsPerModulation
 		stats.TxPacketsPerStatus = s.TxPacketsPerStatus
+		stats.Location = s.Location
```

Comment:
Are you sure about this? Please note that `s` comes from `conn.stats.ExportStats()`. This function is defined here:
https://github.com/brocaar/chirpstack-gateway-bridge/blob/master/internal/backend/stats/stats.go#L106

This function does not track the location, and this will always set `stats.Location` to `nil`. The reason we are setting stats using the `ExportStats` is that the ChirpStack Gateway Bridge is aggregating more stats than provided by the Semtech UDP Packet Forwarder.

The `stats` variable that is passed to `handleStats` comes from here:
https://github.com/brocaar/chirpstack-gateway-bridge/blob/2eeb1d76883e86172b00a84f9c61a29585b9cb61/internal/backend/semtechudp/packets/push_data.go#L53

This function maps the Semtech UDP Packet Forwarder JSON to the ChirpStack data-structure and sets the location if available. Thus if there is a location available, then `stats` already contains this and `stats.Location = s.Location` would cause it to unset.

Reply:
Well, thanks, please allow me some time to re-check my conclusions about this point - I'll come back soon!

Reply:
@brocaar seems you are right, thanks, this is mistake of mine. I removed the line from branch. Probably it's better to squash commits if/when merging.